hawk/app/models/report.rb

```diff
@@ -73,39 +229,78 @@ def find(id)
     def all
       report_files.values.map do |file|
         parse(file)
-      end.sort_by(&:order)
+      end.sort_by(&:name)
     end
 
-    def report_files
-      files = report_path.children.select do |file|
-        if file.extname == ".bz2"
-          file
-        end
-      end
+    def report_name(file)
+      n = file
+      n = n.basename(n.extname)
+      n = n.basename(n.extname) if n.extname == ".tar"
+      n.to_s
+    end
 
-      {}.tap do |result|
-        files.each do |file|
-          hash = Digest::SHA1.hexdigest(
-            file.basename(".tar.bz2").to_s
-          )
 
-          result[hash] = file
+    def report_id(name)
+      Digest::SHA1.hexdigest(name)[0..8]
+    end
+
+    def report_files
+      basenames = [".bz2", ".gz", ".xz"]
+      files = report_path.children.select { |file| file if basenames.include? file.extname }
+      {}.tap do |ret|
+        files.each do |file|
+          id = report_id(report_name(file))
+          ret[id] = file
         end
       end
     end
 
     def report_file(name)
-      report_files[name]
+      reports = report_files
+      Rails.logger.debug "Reports: #{reports}, looking for #{name}"
+      reports[name]
     end
 
     def report_path
-      @report_path ||= Rails.root.join("tmp", "explorer")
+      @report_path ||= Rails.root.join("tmp", "reports")
+      @report_path.mkpath unless @report_path.directory?
+      @report_path
+    end
+  end
 
-      unless @report_path.directory?
-        @report_path.mkpath
+  class Upload < Tableless
+    attribute :upload, ActionDispatch::Http::UploadedFile
+
+    validate do |record|
+      unless record.upload.content_type == "application/x-bzip" ||
+             record.upload.content_type == "application/x-xz" ||
+             record.upload.content_type == "application/x-gz"
```

Comment:
Style/MultilineOperationIndentation: Use 4 (not 7) spaces for indenting a condition in an `unless` statement spanning multiple lines.